ui/components/app/alerts-system/utils.ts

```diff
@@ -0,0 +1,125 @@
+import {
+  BackgroundColor,
+  Severity,
+} from '../../../helpers/constants/design-system';
+import { useI18nContext } from '../../../hooks/useI18nContext';
+import {
+  REASON_TO_DESCRIPTION_TKEY,
+  REASON_TO_TITLE_TKEY,
+} from '../../../pages/confirmations/components/security-provider-banner-alert/blockaid-banner-alert/blockaid-banner-alert';
+import { SecurityAlertResponse } from '../../../pages/confirmations/types/confirm';
+import { BannerAlertSeverity } from '../../component-library';
+
+/**
+ * Returns the background color based on the severity level.
+ *
+ * @param severity - The severity level.
+ * @returns The background color corresponding to the severity level.
+ */
+export function getSeverityBackground(severity: Severity): BackgroundColor {
+  switch (severity) {
+    case Severity.Danger:
+      return BackgroundColor.errorMuted;
+    case Severity.Warning:
+      return BackgroundColor.warningMuted;
+    default:
+      return BackgroundColor.primaryMuted;
+  }
+}
+
+/**
+ * Returns the highest  severity from an array of alerts.
+ *
+ * @param alerts - An array of alerts.
+ * @returns The highest  severity.
+ */
+export function getHighestSeverity(alerts: Alert[]): AlertSeverity {
+  return alerts.reduce((highestSeverity: AlertSeverity, alert: Alert) => {
+    if (alert.severity === Severity.Danger) {
+      return Severity.Danger;
+    } else if (
+      alert.severity === Severity.Warning &&
+      highestSeverity !== Severity.Danger
+    ) {
+      return Severity.Warning;
+    }
+    return highestSeverity;
+  }, Severity.Info);
+}
+
+/**
+ * Converts the severity of a banner alert to the corresponding BannerAlertSeverity.
+ *
+ * @param severity - The severity of the banner alert.
+ * @returns The corresponding BannerAlertSeverity.
+ */
+export function getBannerAlertSeverity(
+  severity: Severity,
+): BannerAlertSeverity {
+  switch (severity) {
+    case Severity.Danger:
+      return BannerAlertSeverity.Danger;
+    case Severity.Warning:
+      return BannerAlertSeverity.Warning;
+    default:
+      return BannerAlertSeverity.Info;
+  }
+}
+
+/**
+ * Returns the corresponding AlertSeverity based on the provided BlockaidResultType.
+ *
+ * @param severity - The BlockaidResultType to determine the AlertSeverity for.
+ * @returns The AlertSeverity corresponding to the provided BlockaidResultType.
+ */
+export function getProviderAlertSeverity(
+  severity: BlockaidResultType,
+): AlertSeverity {
+  switch (severity) {
+    case BlockaidResultType.Malicious:
+      return Severity.Danger;
+    case BlockaidResultType.Warning:
+      return Severity.Warning;
+    default:
+      return Severity.Info;
+  }
+}
+
+/**
+ * Normalizes a security alert response into an Alert object.
+ *
+ * @param securityAlertResponse - The security alert response to normalize.
+ * @param t - The translation function.
+ * @returns The normalized Alert object.
+ */
+export function providerAlertNormalizer(
```

Comment:
Same here, this is for a concrete usage of an alert, so we don't want to couple it within the alert system itself.

Reply:
Moved `providerAlertNormalizer` as well to `hooks/alerts/utils.ts`.